src/python/pants/backend/python/subsystems/python_setup.py

```diff
@@ -160,3 +168,64 @@ def _failsafe_parse(self, requirement):
       return Requirement.parse(requirement, replacement=False)
     except TypeError:
       return Requirement.parse(requirement)
+
+  @classmethod
+  def expand_interpreter_search_paths(cls, interpreter_search_paths, pyenv_root_func=None):
+    special_strings = {
+      '<PEXRC_PATH>': cls.get_pex_python_paths,
+      '<PATH>': cls.get_environment_paths,
+      '<PYENV>': lambda: cls.get_pyenv_paths(pyenv_root_func=pyenv_root_func)
+    }
+    expanded = []
+    for s in interpreter_search_paths:
```

Comment:
In accordance with the previous state of the code, `PEXRC_PATH` should take precedent no matter what. If it is set, it is the first thing that Pants searches for. This is intended to provide alignment with the underlying pex library's behavior when these vars are present in a .pexrc somewhere. I think it wouldn't be unreasonable to always prepend the PEXRC_PATH func results to the search paths if the return value is not empty. Thoughts?

Reply:
Keep in mind that the current behavior doesn't add the pexrc values to the option values - it overrides the option values entirely. And that's exactly what I'm trying to alter with this change. I don't think it makes sense to silently ignore a configured option because of the accidental presence of a pexrc file on a system. If a repo wants to always prepend the pexrc path, they can do this via the option, but there's no reason to force this.

Note that it's not a matter of precedence. The order of these paths is ignored - the interpreter cache looks for interpreters at every entry in the path, and chooses the lowest version that's compatible with all constraints.

Reply:
> the interpreter cache looks for interpreters at every entry in the path, and chooses the lowest version that's compatible with all constraints. 
This is a result of these changes afaict. Previously if pexrc is set, you are getting the lowest compatible interpreter within that setting, with no possibility of a system/other interpreter sliding underneath the pexrc ones at selection time. 

How would you feel about defaulting to only the <PEXRC> variable so the PATH interpreters cannot be selected as the min interpreter? Also, it would be great to add logging for the case where pants detects PEXRC but ignores it in favor of custom paths (in the event that the user is unaware of Pants respects it to ensure maximum alignment for build/runtime interpreters).

Reply:
If you want the previous behavior all you have to do is set the value of `--interpreter-search-paths` to `[<PEXRC>]`. The problem with having that as the default is that then Pants won't work out of the box for anyone who doesn't have a PEXRC set.  This feels like something a repo needs to consciously choose. If I understand the pexrc feature correctly, it's for hermeticity? But the old behavior wasn't truly hermetic, in that if the user didn't have a pexrc, they'd silently fall back to their PATH. It feels like if a repo demands hermeticity then interpreter selection should fail without a pexrc, no?

Reply:
I see what you are saying. The older feature was to ensure hermeticity _if_ pexrc is set and give that precedence over everything else. This was more of an implicit feature than something the user consciously chose to do, like "nice thing that Pants would make sure happens on your behalf."